docs/core/compatibility/sdk/6.0/csharp-template-code.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Breaking change: C# code in templates not supported by earlier versions"
+description: Learn about the breaking change in .NET 6 where project templates use new C# features that aren't available in previous versions.
+ms.date: 07/30/2021
+---
+# C# code in templates not supported by earlier versions
+
+Starting in .NET 6, the project templates that ship with the .NET SDK use the latest C# language features. The following language features are used in template-generated code or enabled by default in the project:
+
+- [Top-level statements](../../../../csharp/fundamentals/program-structure/top-level-statements.md)
+- [Global using directives](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/GlobalUsingDirective.md)
+- [File-scoped namespaces](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/file-scoped-namespaces.md)
+- [Target-typed new expressions](/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new)
+- [Nullable reference types](../../../../csharp/nullable-references.md)
+- [`async` Main method](/dotnet/csharp/language-reference/proposals/csharp-7.1/async-main)
+
+Some of the latest C# language features are not supported by previous target frameworks, so you might experience issues in the following scenarios:
+
+- When you use a .NET 6 template and then change the target framework to a previous version.
+- When you use a .NET 6 template and then multi-target by adding a previous target framework version.
+
+## Version introduced
+
+.NET SDK 6.0.100 Preview 7
+
+## Old behavior
+
+In previous .NET versions, you can change the target framework to a previous version or add an additional version without having to change the C# code created by the project template.
+
+## New behavior
+
+Starting in .NET 6, when you change or add a target framework, you may need to change the C# code generated by the template to avoid using unsupported language features. The project won't build without these changes. The compiler errors and warnings should usually guide you on how to change the generated code to make it compatible with new target framework.
```

Comment:
```suggestion
Starting in .NET 6, when you change or add a target framework, you may need to change the C# code generated by the template to avoid using unsupported language features. The project won't build without these changes. The compiler errors and warnings usually guide you on how to change the generated code to make it compatible with new target framework.
```